core/src/KokkosExp_MDRangePolicy.hpp

```diff
@@ -314,7 +314,38 @@ struct MDRangePolicy : public Kokkos::Impl::PolicyTraits<Properties...> {
   }
   bool impl_tune_tile_size() const { return m_tune_tile_size; }
 
+  tile_type tile_size_recommended() const {
+    auto properties             = Impl::get_tile_size_properties(m_space);
+    tile_type default_tile_size = {};
+
+    for (std::size_t i = 0; i < default_tile_size.size(); ++i) {
+      default_tile_size[i] = properties.default_tile_size;
+    }
+
+    int last_rank = (inner_direction == Iterate::Right) ? rank - 1 : 0;
+    default_tile_size[last_rank] = tile_length_last_rank(
+        properties, m_upper[last_rank] - m_lower[last_rank]);
+    return default_tile_size;
+  }
```

Comment:
This might as well just return `m_tile` but I can also see that we want to match the behavior of `team_size_recommended`.

Reply:
agreed